Classes/Player/AssetBuilder.swift

```diff
@@ -11,15 +11,7 @@ import AVFoundation
 
 class AssetBuilder {
     
-    let mediaEntry: MediaEntry
-    var assetHandler: AssetHandler?
-    
-    init(mediaEntry: MediaEntry) {
-        self.mediaEntry = mediaEntry
-    }
-
-    func getPreferredMediaSource() -> (MediaSource, AssetHandler.Type)? {
-        
+    static func getPreferredMediaSource(fromMediaEntry mediaEntry: MediaEntry) -> (MediaSource, AssetHandler.Type)? {
```

Comment:
I think it's more Swifty for the label to be "for". Then when you call the method, it's `getPreferredMediaSource(for: mediaEntry)` instead of `getPreferredMediaSource(fromMediaEntry: mediaEntry)`.